docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,449 @@
+    edisp = Map.read("stacked/edisp.fits")
+    psf = Map.read("stacked/psf.fits")
+
+    dataset = MapDataset(
+        counts=counts,
+        exposure=exposure,
+        edisp=edisp,
+        psf=psf,
+        mask=mask,
+        model=model,
+        background=background_model,
+        likelihood="cash"
+        )
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+The edisp and psf maps can be defined on a coarse spatial geometry. The counts,
+exposure and background map are supposed to be on the same spatial geometry,
+chosen by the user for the analysis.
+
+The `MapDataset` implements at least the following methods:
+
+.. code::
+
+    dataset = MapDataset()
+
+    # the map dataset might require some precomputations
+    dataset.setup()?
+
+    dataset.npred()
+    dataset.likelihood_per_bin()
+    dataset.likelihood(mask)
+
+The `.setup()` method creates cutouts of the exposure map for the individual
+model components, assigns the correspoding IRFs to the model component and
+bundles all into `SourceIRFModel` or `ModelEvaluator()` classes. The list of model
+evaluators is cached on the `MapDataset` and used later to efficiently compute
+the predicted number of counts.
+
+The `mask` argument to the `.likelihood` method allows to set a global fitting
+mask, when mutiple `MapDataset` are fitted jointly.
+
+
+`MapDatasetOnOff`
+----------------
+For on-off based analyses a `MapDatasetOnOff` class could be introduced.
+
+.. code ::
+
+    from gammapy.cube import MapDatasetOnOff
+
+    model = SourceModel.read("model.yaml")
+
+    counts_on = Map.read("stacked/counts_on.fits")
+    counts_off = Map.read("stacked/counts_off.fits")
+    alpha = Map.read("stacked/alpha.fits")
+
+    edisp = Map.read("stacked/edisp.fits")
+    psf = Map.read("stacked/psf.fits")
+
+    dataset_onoff = MapDatasetOnOff(
+        counts_on=counts_on,
+        counts_off=counts_off,
+        alpha=alpha,
+        edisp=edisp,
+        psf=psf,
+        mask=mask,
+        model=model,
+        likelihood="wstat"
+        )
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+
+`SpectrumDataset`
+-----------------
+
+For spectral analysis we propose to introduce a `SpectrumDataset`:
+
+.. code ::
+
+    from gammapy.spectrum import SpectrumDataset
+
+    model = SpectralModel()
+    background_model = SpectralBackgroundModel.read()
+    edisp = EnergyDispersion.read()
+
+    counts = Spectrum.read()
+    exposure = Spectrum.read()
+    mask = Spectrum.read()
+
+    dataset = SpectrumDataset(
+        counts=counts,
+        exposure=exposure,
+        model=model,
+        background_model=background_model,
+        mask=mask,
+        edisp=edisp,
+        likelihood="cash"
+        )
+
+    dataset.npred()
+    dataset.likelihoop_per_bin()
+    dataset.likelihood(mask)
+
+The `SpectrumDatasets` with a parametric background model will be introduced first.
+The existing `SpectrumObservation` can be refactored later in a `SpectrumOnOffDataset`
+or `PHASpectrumDatasets`.
```

Comment:
Same suggestions: don't commit to separate class for on/off in the PIG, mention options. But as you prefer, OK to merge either way.
Suggest to remove the "or PHASpectrumDatasets" - PHA is a cryptic term, shouldn't be in the Gammapy name / design, it's just a legacy thing that we support as I/O format.